arch/arm/src/cxd56xx/cxd56_sysctl.c

```diff
@@ -127,13 +164,24 @@ int cxd56_sysctlcmd(uint8_t id, uint32_t data)
   ret = cxd56_iccsend(CXD56_PROTO_SYSCTL, &msg, SYSCTL_TIMEOUT);
   if (ret < 0)
     {
+      sysctl_semgive(&g_exc);
       _err("Timeout.\n");
       return ret;
     }
 
   /* Wait for reply message from system CPU */
 
-  sysctl_semtake(&g_sync);
+  do
+    {
+      ret = sysctl_semtake(&g_sync);
+
+      /* The only expected error is ECANCELED which would occur if the
+       * calling thread were canceled.
+       */
+
+      DEBUGASSERT(ret == OK || ret == -ECANCELED);
+    }
+  while (ret < 0);
```

Comment:
It there any reason for this loop here?  This loop basically ignores cancellation requests (and loses the cancellation status permanently).  Why can't you just return following the same logic as at lines 167-169.